src/Utils/FileSystemUtils.php

```diff
@@ -9,13 +9,47 @@ class FileSystemUtils
 {
     public function collectFilePathsRecursively($rootPath, $pattern)
     {
-        $paths = $this->collectPathsRecursively($rootPath, $pattern);
+        if (strpos($rootPath, '*') !== false || strpos($rootPath, '?') !== false) {
+            $paths = $this->collectGlobs($rootPath, $pattern);
+        } else {
+            $paths = $this->collectPathsRecursively($rootPath, $pattern);
+        }
 
         return array_filter($paths, function ($item) {
             return is_file($item);
         });
     }
 
+    public function collectGlobs($rootPath, $pattern)
```

Comment:
Should i follow te `Recursively` naming schema here?
It does not explicitly use a `RecursiveIterator` like `collectPathsRecursively` does, but it does call `collectPathsRecursively`